api/client/export.go

```diff
@@ -40,6 +38,39 @@ func (cli *DockerCli) CmdExport(args ...string) error {
 	}
 	defer responseBody.Close()
 
+	var tmpFile *os.File
+	if *outfile != "" {
+		tmpFile, err = ioutil.TempFile(filepath.Dir(*outfile), ".docker_export_")
+		if err != nil {
+			return err
+		}
+		output = tmpFile
+		tmpPath := tmpFile.Name()
+
+		defer func() {
+			tmpFile.Close()
+			if _, err := os.Stat(tmpPath); err == nil {
+				os.Remove(tmpPath)
+			}
+		}()
+	}
 	_, err = io.Copy(output, responseBody)
-	return err
+
+	if err != nil {
+		return err
+	}
+
+	if *outfile != "" {
+		if err = os.Remove(*outfile); err != nil && !os.IsNotExist(err) {
```

Comment:
@tonistiigi Oh Thanks. i will remove `Remove` function. how about write new function in the utils.go?